### PR TITLE
Fix check for (neg) zero for fpclass emulation

### PR DESCRIPTION
**Pull request by asudarsa**
We should compare not only to zero integer, but also 'negated' zero.

This PR is built in top of: Fix check for (neg) zero for fpclass emulation #2151

### CODE REVIEW

**Comment by asudarsa** [comment]
Hi @svenvh and @LU-JOHN 

Sorry for inconvenience. Can we please continue the review of PR #2151 here? My last commit introduces a bitwise and as suggested by John.

Thanks

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -4434,17 +4434,42 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
...
-      auto *ZeroConst = transValue(
-          Constant::getIntegerValue(IntOpLLVMTy, APInt::getZero(BitSize)), BB);
-      auto *TestIsZero =
-          BM->addCmpInst(OpIEqual, ResTy, BitCastToInt, ZeroConst, BB);
-      if (FPClass & fcPosZero && FPClass & fcNegZero)
+      if (FPClass & fcPosZero && FPClass & fcNegZero) {
+        APInt ZeroInt = APInt::getZero(BitSize);
+        auto *ZeroConst =
+            transValue(Constant::getIntegerValue(IntOpLLVMTy, ZeroInt), BB);
+        APInt SignedMaxInt = APInt::getSignedMaxValue(BitSize);
```
Comment (LU-JOHN) [inline lib/SPIRV/SPIRVWriter.cpp:4457]: Rename variable and/or add comment that this is used as a mask to mask-off floating-point sign-bit.   It is coincidentally signed-max but that is not the essential property in this code.

File: test/llvm-intrinsics/fpclass.ll [hunk 2]
```diff
@@ -21,6 +21,8 @@
 ; CHECK-SPIRV-DAG: Constant [[#Int32Ty]] [[#QNanBitConst:]] 2143289344
 ; CHECK-SPIRV-DAG: Constant [[#Int32Ty]] [[#MantissaConst:]] 8388607
 ; CHECK-SPIRV-DAG: Constant [[#Int32Ty]] [[#ZeroConst:]] 0
+; CHECK-SPIRV-DAG: Constant [[#Int32Ty]] [[#MaxConst:]] 2147483647
```
Comment (LU-JOHN) [inline test/llvm-intrinsics/fpclass.ll:24]: Perhaps a better name than MaxConst would be MaskToClearFPSign.

File: lib/SPIRV/SPIRVWriter.cpp (hunk 1 above)
Comment (asudarsa) [reply lib/SPIRV/SPIRVWriter.cpp]: Thanks. I changed the variable names.

**Review by vmaksimo — APPROVED** [review]

**Review by svenvh — APPROVED** [review]

**Review by LU-JOHN — APPROVED** [review]

**Comment by asudarsa** [comment]
Thanks everyone for review.

**Pull request merged**